osu.Game/Overlays/SocialOverlay.cs

```diff
@@ -166,20 +170,68 @@ private void recreatePanels(PanelDisplayStyle displayStyle)
 
             LoadComponentAsync(newPanels, f =>
             {
-                if (panels != null)
-                    ScrollFlow.Remove(panels);
-
+                loading.Hide();
                 ScrollFlow.Add(panels = newPanels);
             });
         }
 
         private void updateUsers(IEnumerable<User> newUsers)
         {
-            Users = newUsers;
-            loading.Hide();
+            loading.Show();
+
+            var sortDirection = Filter.DisplayStyleControl.Dropdown.Current.Value;
+
+            IEnumerable<User> sortedUsers = newUsers;
+
+            if (sortedUsers.Any())
+            {
+                switch (Filter.Tabs.Current.Value)
+                {
+                    case SocialSortCriteria.Location:
+                        switch (sortDirection)
+                        {
+                            case SortDirection.Ascending:
```

Comment:
why not use `.Reverse()` for ascending/descending rather than duplicating this stuff? or if you want to avoid using `Reverse` (potentially requires more iteration than otherwise), store the predicate to a variable and at very least only have the `OrderBy` / `OrderByDescending` statements once.